grails-app/services/org/pih/warehouse/data/LocationDataService.groovy

```diff
@@ -50,7 +56,7 @@ class LocationDataService {
                 command.errors.reject("Row ${index + 1}: Active field is incorrectly filled. Please fill with TRUE or FALSE")
             }
 
-            if (params.locationNumber && Location.findByLocationNumber(params.locationNumber)) {
+            if (params.locationNumber && Location.findByLocationNumber(params.locationNumber) && !(params.id && params.locationNumber == location?.locationNumber)) {
```

Comment:
Ok here's the location number part. Can you add a comment to explain what's happening?

Reply:
Ok, I'll add a comment.